sky/provision/runpod/utils.py

```diff
@@ -142,23 +144,45 @@ def launch(name: str, instance_type: str, region: str, disk_size: int,
     if ports is not None:
         custom_ports_str = ''.join([f'{p}/tcp,' for p in ports])
 
-    new_instance = runpod.runpod.create_pod(
-        name=name,
-        image_name=image_name,
-        gpu_type_id=gpu_type,
-        cloud_type=cloud_type,
-        container_disk_in_gb=disk_size,
-        min_vcpu_count=4 * gpu_quantity,
-        min_memory_in_gb=gpu_specs['memoryInGb'] * gpu_quantity,
-        gpu_count=gpu_quantity,
-        country_code=region,
-        ports=(f'22/tcp,'
-               f'{custom_ports_str}'
-               f'{constants.SKY_REMOTE_RAY_DASHBOARD_PORT}/http,'
-               f'{constants.SKY_REMOTE_RAY_PORT}/http'),
-        support_public_ip=True,
-        docker_args=
-        f'bash -c \'echo {encoded} | base64 --decode > init.sh; bash init.sh\'')
+    docker_args = (f'bash -c \'echo {encoded} | base64 --decode > init.sh; '
+                   f'bash init.sh\'')
+    ports = (f'22/tcp,'
+             f'{custom_ports_str}'
+             f'{constants.SKY_REMOTE_RAY_DASHBOARD_PORT}/http,'
+             f'{constants.SKY_REMOTE_RAY_PORT}/http')
+
+    if preemptible is None or not preemptible:
+        new_instance = runpod.runpod.create_pod(
+            name=name,
+            image_name=image_name,
+            gpu_type_id=gpu_type,
+            cloud_type=cloud_type,
+            container_disk_in_gb=disk_size,
+            min_vcpu_count=4 * gpu_quantity,
+            min_memory_in_gb=gpu_specs['memoryInGb'] * gpu_quantity,
+            gpu_count=gpu_quantity,
+            country_code=region,
+            ports=ports,
+            support_public_ip=True,
+            docker_args=docker_args,
+        )
+    else:
+        new_instance = create_spot_pod(
+            name=name,
+            image_name=image_name,
+            gpu_type_id=gpu_type,
+            cloud_type=cloud_type,
+            bid_per_gpu=bid_per_gpu,
+            container_disk_in_gb=disk_size,
+            volume_in_gb=disk_size,
+            min_vcpu_count=4 * gpu_quantity,
+            min_memory_in_gb=gpu_specs['memoryInGb'] * gpu_quantity,
+            gpu_count=gpu_quantity,
+            country_code=region,
+            ports=ports,
+            support_public_ip=True,
+            docker_args=docker_args,
+        )
```

Comment:
Just wondering if we should just keep the graphql implementation for creating pod, i.e. use the same graphql implementation for on-demand as well, to make the code structure cleaner.

Reply:
There are input fields defined https://graphql-spec.runpod.io/#definition-PodRentInterruptableInput and https://graphql-spec.runpod.io/#definition-PodFindAndDeployOnDemandInput.

Although some of the fields are not currently being used in runpod python API, I suggest keeping both the create spot pod and create pod implementations. This will make it easier to accommodate future changes and customer needs.

Reply:
update a bit to make it cleaner

Reply:
Ahh, I see. It should be fine then. Let's keep the current two way of implementation